src/pl/ogladajanime/src/eu/kanade/tachiyomi/animeextension/pl/ogladajanime/OgladajAnime.kt

```diff
@@ -0,0 +1,293 @@
+import eu.kanade.tachiyomi.animesource.ConfigurableAnimeSource
+import eu.kanade.tachiyomi.animesource.model.AnimeFilterList
+import eu.kanade.tachiyomi.animesource.model.SAnime
+import eu.kanade.tachiyomi.animesource.model.SEpisode
+import eu.kanade.tachiyomi.animesource.model.Video
+import eu.kanade.tachiyomi.animesource.online.ParsedAnimeHttpSource
+import eu.kanade.tachiyomi.lib.cdaextractor.CdaPlExtractor
+import eu.kanade.tachiyomi.lib.dailymotionextractor.DailymotionExtractor
+import eu.kanade.tachiyomi.lib.mp4uploadextractor.Mp4uploadExtractor
+import eu.kanade.tachiyomi.lib.sibnetextractor.SibnetExtractor
+import eu.kanade.tachiyomi.lib.vkextractor.VkExtractor
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import okhttp3.FormBody
+import okhttp3.Headers
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import uy.kohesive.injekt.injectLazy
+
+class OgladajAnime : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
+
+    override val name = "OgladajAnime"
+
+    override val baseUrl = "https://ogladajanime.pl"
+
+    override val lang = "pl"
+
+    override val supportsLatest = true
+
+    private val json: Json by injectLazy()
+
+    private val apiHeaders = Headers.Builder()
+        .set("Accept", "application/json, text/plain, */*")
+        .set("Referer", "$baseUrl/")
+        .set("Origin", baseUrl)
+        .set("Accept-Language", "pl,en-US;q=0.7,en;q=0.3")
+        .set("Host", "ogladajanime.pl")
+        .build()
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    // ============================== Popular ===============================
+    override fun popularAnimeRequest(page: Int): Request {
+        return GET("$baseUrl/search/page/$page", headers)
+    }
+    override fun popularAnimeSelector(): String = "div#anime_main div.card.bg-white"
+
+    override fun popularAnimeFromElement(element: Element): SAnime {
+        return SAnime.create().apply {
+            setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+            thumbnail_url = element.selectFirst("img")?.attr("data-srcset")
+            title = element.selectFirst("h5.card-title > a")!!.text()
+        }
+    }
+    override fun popularAnimeNextPageSelector(): String = "section:has(div#anime_main)" // To nie działa zostało to tylko dlatego by ładowało ale na końcu niestety wyskakuje ze "nie znaleziono" i tak zostaje zamiast zniknać możliwe ze zle fetchuje.
+
+    // =============================== Latest ===============================
+
+    override fun latestUpdatesRequest(page: Int): Request = GET("$baseUrl/search/new/$page")
```

Comment:
```suggestion
    override fun latestUpdatesRequest(page: Int): Request = GET("$baseUrl/search/new/$page", headers)
```

add `headers` to `GET` elsewhere too

Reply:
My mistake I did not read about it above with understanding D: